### DISCUSSION TITLE
movingbase and rover

### DISCUSSION THREAD

**Discussion by deavn2236** [semuconsulting/pygnssutils#77] [Q&A]
The program you provided is very helpful for GNSS resolution. I have a question to ask.
I'm using the ZED-F9P for moving and rover positioning. I want to achieve positioning and orientation for a small vehicle with it. I use two ZED-F9Ps for interaction, as shown in the picture. How can I use this library to obtain GNSS position and the corresponding orientation of the two boards

https://content.u-blox.com/sites/default/files/documents/ZED-F9P-MovingBase_AppNote_UBX-19009093.pdf
![image](https://github.com/user-attachments/assets/71d3ceae-4095-4b7c-904d-4d4d4daf952e)

![image](https://github.com/user-attachments/assets/b1670d10-1c0d-4d82-bc6c-e584602e7329)

**Comment by semuadmin** [comment]
Hi @deavn2236, 

There's no 'off-the-shelf' solution for this in `pygnssutils`, but assuming you're familiar with basic F9P UBX configuration, enabling and monitoring the UBX-NAV-RELPOSNED message in the rover receiver will give you the relative orientations of base and rover.

FYI the "[Rover Plot](https://github.com/semuconsulting/PyGPSClient/blob/master/src/pygpsclient/rover_frame.py)" widget in the [PyGPSClient](https://github.com/semuconsulting/PyGPSClient) GUI application does exactly this and presents the information the form of a location plot (the centre of the plot represents the base and the orange dots represent the rover vector and trace):

![roverplot](https://raw.githubusercontent.com/semuconsulting/PyGPSClient/master/images/rover_widget.png)

Suggest reading the u-blox interface documentation on the UBX-NAV-RELPOSNED message type e.g.:

https://content.u-blox.com/sites/default/files/documents/u-blox-F9-HPG-L1L5-1.40_InterfaceDescription_UBX-23006991.pdf

Hope this helps.
